packages/osd-monaco/src/xjson/lexer_rules/opensearchppl.ts

```diff
@@ -0,0 +1,529 @@
+  'ASCII',
+  'LOCATE',
+  'REPLACE',
+  'REVERSE',
+  'CAST',
+
+  // BOOL FUNCTIONS
+  'LIKE',
+  'ISNULL',
+  'ISNOTNULL',
+
+  // FLOWCONTROL FUNCTIONS
+  'IFNULL',
+  'NULLIF',
+  'IF',
+  'TYPEOF',
+
+  // RELEVANCE FUNCTIONS AND PARAMETERS
+  'MATCH',
+  'MATCH_PHRASE',
+  'MATCH_PHRASE_PREFIX',
+  'MATCH_BOOL_PREFIX',
+  'SIMPLE_QUERY_STRING',
+  'MULTI_MATCH',
+  'QUERY_STRING',
+  'ALLOW_LEADING_WILDCARD',
+  'ANALYZE_WILDCARD',
+  'ANALYZER',
+  'AUTO_GENERATE_SYNONYMS_PHRASE_QUERY',
+  'BOOST',
+  'CUTOFF_FREQUENCY',
+  'DEFAULT_FIELD',
+  'DEFAULT_OPERATOR',
+  'ENABLE_POSITION_INCREMENTS',
+  'ESCAPE',
+  'FLAGS',
+  'FUZZY_MAX_EXPANSIONS',
+  'FUZZY_PREFIX_LENGTH',
+  'FUZZY_TRANSPOSITIONS',
+  'FUZZY_REWRITE',
+  'FUZZINESS',
+  'LENIENT',
+  'LOW_FREQ_OPERATOR',
+  'MAX_DETERMINIZED_STATES',
+  'MAX_EXPANSIONS',
+  'MINIMUM_SHOULD_MATCH',
+  'OPERATOR',
+  'PHRASE_SLOP',
+  'PREFIX_LENGTH',
+  'QUOTE_ANALYZER',
+  'QUOTE_FIELD_SUFFIX',
+  'REWRITE',
+  'SLOP',
+  'TIE_BREAKER',
+  'TYPE',
+  'ZERO_TERMS_QUERY',
+
+  // SPAN KEYWORDS
+  'SPAN',
+  'MS',
+  'S',
+  'M',
+  'H',
+  'W',
+  'Q',
+  'Y',
+];
+
+const builtinFunctions = [
+  'ABS',
+  'CBRT',
+  'CEIL',
+  'CEILING',
+  'CONV',
+  'CRC32',
+  'E',
+  'EXP',
+  'FLOOR',
+  'LN',
+  'LOG',
+  'LOG10',
+  'LOG2',
+  'MOD',
+  'PI',
+  'POSITION',
+  'POW',
+  'POWER',
+  'RAND',
+  'ROUND',
+  'SIGN',
+  'SQRT',
+  'TRUNCATE',
+  'ACOS',
+  'ASIN',
+  'ATAN',
+  'ATAN2',
+  'COS',
+  'COT',
+  'DEGREES',
+  'RADIANS',
+  'SIN',
+  'TAN',
+  'ADDDATE',
+  'ADDTIME',
+  'CURDATE',
+  'CURRENT_DATE',
+  'CURRENT_TIME',
+  'CURRENT_TIMESTAMP',
+  'CURTIME',
+  'DATE',
+  'DATEDIFF',
+  'DATE_ADD',
+  'DATE_FORMAT',
+  'DATE_SUB',
+  'DAYNAME',
+  'DAYOFMONTH',
+  'DAYOFWEEK',
+  'DAYOFYEAR',
+  'DAY_OF_MONTH',
+  'DAY_OF_WEEK',
+  'DAY_OF_YEAR',
+  'EXTRACT',
+  'FROM_DAYS',
+  'FROM_UNIXTIME',
+  'GET_FORMAT',
+  'LAST_DAY',
+  'LOCALTIME',
+  'LOCALTIMESTAMP',
+  'MAKEDATE',
+  'MAKETIME',
+  'MONTHNAME',
+  'NOW',
+  'PERIOD_ADD',
+  'PERIOD_DIFF',
+  'SEC_TO_TIME',
+  'STR_TO_DATE',
+  'SUBDATE',
+  'SUBTIME',
+  'SYSDATE',
+  'TIME',
+  'TIMEDIFF',
+  'TIMESTAMP',
+  'TIMESTAMPADD',
+  'TIMESTAMPDIFF',
+  'TIME_FORMAT',
+  'TIME_TO_SEC',
+  'TO_DAYS',
+  'TO_SECONDS',
+  'UNIX_TIMESTAMP',
+  'UTC_DATE',
+  'UTC_TIME',
+  'UTC_TIMESTAMP',
+  'WEEKDAY',
+  'YEARWEEK',
+];
+
+export const lexerRules = {
+  defaultToken: 'invalid',
+  ignoreCase: true,
+  tokenPostfix: '',
+  keywords,
+  builtinFunctions,
+  brackets,
+  tokenizer: {
+    root: [
+      [
+        /[a-zA-Z_$][a-zA-Z0-9_$]*\b/,
+        {
+          cases: {
+            '@keywords': 'keyword',
+            '@builtinFunctions': 'identifier',
+            '@default': 'identifier',
+          },
+        },
+      ],
+      [/[()]/, '@brackets'],
+      [/--.*$/, 'comment'],
+      [/\/\*/, 'comment', '@comment'],
+      [/\/.*$/, 'comment'],
+
+      [/".*?"/, 'string'],
+      [/'.*?'/, 'constant'],
```

Comment:
curious how are these defined? i don't think PPL supports comments, and the string definition is a bit different from sql

Reply:
good catch, i'm not sure why that's in there. best guess is it was part of the boilerplate. i'll look it over and make the needed changes

Reply:
is the boilerplate generated by something or copied from some pre-existing code? or is this written from scratch by this PR? asking because looks like the rules are from some standard but might not completely apply to PPL syntax. probably won't break but just curious

Reply:
i just mean copied from the other monarch lexer rule files, i agree on it not completely applying to ppl